numpy-dev (1.25) deprecation for find_common_type and some __new__

23 new failures from numpy-dev with things like this:

```
FAILED astropy/stats/tests/test_funcs.py::test_kuiper_two_uniform[...] - DeprecationWarning: np.find_common_type is deprecated.  Please use `np.result_type` or `np.promote_types`.
See https://numpy.org/devdocs/release/1.25.0-notes.html and the docs for more information.  (Deprecated NumPy 1.25)
```
```
FAILED astropy/stats/tests/test_funcs.py::test_kuiper_two_nonuniform[...] - DeprecationWarning: np.find_common_type is deprecated.  Please use `np.result_type` or `np.promote_types`.
See https://numpy.org/devdocs/release/1.25.0-notes.html and the docs for more information.  (Deprecated NumPy 1.25)
```
```
FAILED astropy/stats/tests/test_funcs.py::test_detect_kuiper_two_different - DeprecationWarning: np.find_common_type is deprecated.  Please use `np.result_type` or `np.promote_types`.
See https://numpy.org/devdocs/release/1.25.0-notes.html and the docs for more information.  (Deprecated NumPy 1.25)
```
```
FAILED astropy/stats/tests/test_funcs.py::test_fpp_kuiper_two[...] - DeprecationWarning: np.find_common_type is deprecated.  Please use `np.result_type` or `np.promote_types`.
See https://numpy.org/devdocs/release/1.25.0-notes.html and the docs for more information.  (Deprecated NumPy 1.25)
```
```
FAILED astropy/units/tests/test_quantity_non_ufuncs.py::TestLinAlg::test_svd - TypeError: SVDResult.__new__() missing 2 required positional arguments: 'S' and 'Vh'
```
```
FAILED astropy/units/tests/test_quantity_non_ufuncs.py::TestLinAlg::test_qr - TypeError: QRResult.__new__() missing 1 required positional argument: 'R'
```
```
FAILED astropy/units/tests/test_quantity_non_ufuncs.py::TestLinAlg::test_eig
FAILED astropy/units/tests/test_quantity_non_ufuncs.py::TestLinAlg::test_eigh
-  TypeError: EigResult.__new__() missing 1 required positional argument: 'eigenvectors'
```

Example log: https://github.com/astropy/astropy/actions/runs/5018146818/jobs/8997192325?pr=14845

cc @larrybradley and @mhvk

--- a/astropy/units/quantity.py
+++ b/astropy/units/quantity.py
@@ -716,7 +716,10 @@ class Quantity(np.ndarray):
         if isinstance(result, (tuple, list)):
             if out is None:
                 out = (None,) * len(result)
-            return result.__class__(
+            # Some np.linalg functions return namedtuple, which is handy to access
+            # elements by name, but cannot be directly initialized with an iterator.
+            result_cls = getattr(result, "_make", result.__class__)
+            return result_cls(
                 self._result_as_quantity(result_, unit_, out_)
                 for (result_, unit_, out_) in zip(result, unit, out)
             )

